src/sentry/web/frontend/auth_organization_id_login.py

```diff
@@ -0,0 +1,27 @@
+from django.urls import reverse
+from django.views.decorators.cache import never_cache
+
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.utils.auth import is_valid_redirect
+from sentry.web.frontend.auth_organization_login import AuthOrganizationLoginView
+
+
+class AuthOrganizationIdentifierLoginView(AuthOrganizationLoginView):
+    @never_cache
+    def handle(self, request, organization_id):
```

Comment:
If `request.subdomain` is not `None`, then it should have precedence over `organization_id`, and we should do a redirect to the correct `organization_id`.